sdk/program/src/stake_history.rs

```diff
@@ -15,6 +15,42 @@ pub struct StakeHistoryEntry {
     pub deactivating: u64, // requested to be cooled down, not fully deactivated yet
 }
 
+impl StakeHistoryEntry {
+    pub fn fully_active(effective: u64) -> Self {
+        Self {
+            effective,
+            ..Self::default()
+        }
+    }
+
+    pub fn activating(effective: u64, activating: u64) -> Self {
```

Comment:
nit: up to you, but I wonder if it would be clearer to call this `effective_and_activating`, or if just calling it `activating` clearly implies that there may be some fully active.

Reply:
yeah I wanted to shrug this off but I think your intuition about this not being very clear is correct. I changed the method names to `with_*` and so I have a `with_effective_and_activating` method which is pretty clear now.